### PR TITLE
Add riscv64 support

### PR DESCRIPTION
**Pull request by kxxt**
### Summary of the PR

Add riscv64 support.

### Requirements

Before submitting your PR, please make sure you addressed the following
requirements:

- [x] All commits in this PR have Signed-Off-By trailers (with
  `git commit -s`), and the commit message has max 60 characters for the
  summary and max 75 characters for each description line.
- [x] All added/changed functionality has a corresponding unit/integration
  test.
- [x] All added/changed public-facing functionality has entries in the "Upcoming 
  Release" section of CHANGELOG.md (if no such section exists, please create one).
- [ ] Any newly added `unsafe` code is properly documented.

### CODE REVIEW

**Comment by alindima** [comment]
thanks for the contribution. while it overall looks good, we don't have any riscv support in the CI. Without that, I don't think we can merge such PRs, since it would basically result in untested code being released.
I noticed at least one other rust-vmm crate which has open PRs for riscv support also. Is there such planned support?
CC: @andreeaflorescu (mentioning you since you may have an opinion on this)

**Comment by kxxt** [comment]
>  while it overall looks good, 

Thank!

> while it overall looks good, we don't have any riscv support in the CI. 

It looks like there's a PR adding that: https://github.com/rust-vmm/rust-vmm-ci/pull/148

> Without that, I don't think we can merge such PRs, since it would basically result in untested code being released.

I think another possible way could be merging riscv64 support and labeling riscv64 as a community supported and not officially tested architecture in the documentation.

BTW here is a test log on riscv64: https://paste.rs/Oyi0p

**Review by petreeftime — APPROVED** [review]

**Comment by kxxt** [comment]
With the riscv64 CI in place, I think it is time to revive this PR. (Thanks to @TimePrinciple)

**Comment by TimePrinciple** [comment]
Could @roypat @rbradford @petreeftime @alindima take a look when convenient 😉

**Review by rbradford — COMMENTED** [review]
Looks fine to me

File: src/syscall_table/aarch64.rs [hunk 1]
```diff
@@ -1,9 +1,9 @@
-// Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// Copyright 2024 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
[the remark this answers is not in the document]
Comment (kxxt) [reply src/syscall_table/aarch64.rs]: Fixed

File: src/syscall_table/riscv64.rs [hunk 2]
```diff
@@ -0,0 +1,311 @@
+// Copyright 2024 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR BSD-3-Clause
```
Comment (roypat) [inline src/syscall_table/riscv64.rs:2]: Mh, I don't think the new files you authored should have Amazon copyright :sweat_smile:

Comment (TimePrinciple) [reply src/syscall_table/riscv64.rs:2]: That's generated from `generate_syscall_tables.sh`, which automatically adds these lines 😂

Comment (kxxt) [reply src/syscall_table/riscv64.rs:2]: > Mh, I don't think the new files you authored should have Amazon copyright 😅

Technically I updated the script that generates this file. IANAL, but I think it's fine to keep that copyright header if the script is under Amazon copyright and my contribution is licensed under the same licenses.

**Review by alindima — APPROVED** [review]
Thanks! LGTM

**Review by roypat — APPROVED** [review]
(rubber stamping based on @alindima's approval)

**Review by rbradford — APPROVED** [review]

**Pull request merged**